### PR TITLE
18AL/18GA/18MEX/18TN: Limit depot train buy to one before phase 4 (#961)

### PR DESCRIPTION
**Pull request by perwestling**
This applies to the following 18xx:
  - 18AL
  - 18GA
  - 18MEX
  - 18TN

When buying train from depot in phase 2 and phase 3
only one train can be bought.

### CODE REVIEW

**Comment by jenf** [comment]
In general the approach looks fine, but I think it'll need to wait until after the steps rewrite gets merged. It'll be a single class then

**Comment by perwestling** [comment]
@jenf and @tobymao: Now this is rewritten to match the new steps so it should be possible to merge.

File: lib/engine/step/g_18_al/train.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
```
Comment (jenf) [inline lib/engine/step/g_18_al/train.rb:1]: Looks good, only real comment is the use of 'g_18_al' when it's common to nearly all of Mark Derrick's games.
How about g_18_markderrick?

or train_buyone_until_phase4.rb?

Comment (perwestling) [reply lib/engine/step/g_18_al/train.rb]: Renamed it to a descriptive name, and put the file in lib/engine/step.

**Pull request merged**